### PR TITLE
Signed cookie

### PR DESCRIPTION
**Pull request by neovintage**
I took a first pass at trying to sign the session id before it ends up in the cookie. The encoding and decoding shouldn't cause any problems with the session_id itself. This should take care of the issue I created earlier: #2 

Let me know what you think.

### CODE REVIEW

**Comment by sdogruyol** [comment]
@neovintage thanks for the pull request.

It looks good but i think it's better to have a  default `secret_key` instead of making it nilable.

**Comment by neovintage** [comment]
@sdogruyol That's fair. That would greatly simply a bunch of stuff in the PR. I think the best approach would be to use empty string `""` as the default. That way between deployments the sessions won't all change on someone if they're not using a secret token. What do you think?

**Comment by sdogruyol** [comment]
@neovintage having `""` sounds legit.

File: README.md [hunk 1]
```diff
@@ -65,6 +65,7 @@ Session.config.cookie_name = "foobar"
 | cookie_name | Name of the cookie that holds the session_id on the client | ```"kemal_sessid"``` |
 | engine | How are the sessions saved on the server? (see section below) | ```Session::FileSystemEngine.new({sessions_dir: "./sessions/"})``` |
 | gc_interval | In which interval should the garbage collector find and delete expired sessions from the server?  | ```Time::Span.new(0, 4, 0)``` (4 minutes)  |
+| secret_token | Used to sign the session ids before theyre saved in the cookie. Default is to not sign cookies. | nil |
```
Comment (sdogruyol) [inline README.md]: This needs to be updated since cookies are always signed

File: src/kemal-session/base.cr [hunk 2]
```diff
@@ -1,19 +1,46 @@
...
 
   def initialize(context : HTTP::Server::Context)
     Session.config.set_default_engine unless Session.config.engine_set?
     id = context.request.cookies[Session.config.cookie_name]?.try &.value
-    if id && id.size == 32
-      # valid
-    else
+    valid = false
+    if id
+      t_id = URI.unescape(id)
```
Comment (sdogruyol) [inline src/kemal-session/base.cr]: How about removing this temporary variable since it's not used anywhere else?

File: src/kemal-session/base.cr [hunk 3]
```diff
@@ -1,19 +1,46 @@
...
+      parts = t_id.split("--")
```
Comment (sdogruyol) [inline src/kemal-session/base.cr]: `URI.unescape(id).split("--")`

File: src/kemal-session/base.cr [hunk 4]
```diff
@@ -1,19 +1,46 @@
...
+      if parts.size == 2
+        new_val = self.class.sign_value(parts[0])
+        id = parts[0]
+        valid = true if new_val == parts[1] && parts[0].size == 32
+      end
+    end
+
+    if !valid || id.nil?
```
Comment (sdogruyol) [inline src/kemal-session/base.cr]: `id.nil? || !valid`

**Comment by neovintage** [comment]
@sdogruyol rolled in feedback. Let me know what you think.

**Pull request merged**